packages/devtools_app/test/test_infra/scenes/standalone_ui/property_editor_sidebar.dart

```diff
@@ -0,0 +1,158 @@
+class PropertyEditorSidebarScene extends Scene {
+  @override
+  Widget build(BuildContext context) {
+    return const _PropertyEditorSidebar();
+  }
+
+  @override
+  String get title => '$PropertyEditorSidebarScene';
+
+  @override
+  Future<void> setUp() async {
+    setStagerMode();
+    setGlobal(
+      DevToolsEnvironmentParameters,
+      ExternalDevToolsEnvironmentParameters(),
+    );
+    setGlobal(DTDManager, MockDTDManager());
+    setGlobal(IdeTheme, IdeTheme());
+    setGlobal(PreferencesController, PreferencesController());
+  }
+}
+
+class _PropertyEditorSidebar extends StatefulWidget {
+  const _PropertyEditorSidebar();
+
+  @override
+  State<_PropertyEditorSidebar> createState() => _PropertyEditorState();
+}
+
+class _PropertyEditorState extends State<_PropertyEditorSidebar> {
+  Stream<String>? clientLog;
+  DartToolingDaemon? clientDtd;
+  SimulatedEditor? editor;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: themeFor(
+        isDarkTheme: false,
+        ideTheme: _ideTheme(const VsCodeTheme.light()),
+        theme: ThemeData(useMaterial3: true, colorScheme: lightColorScheme),
+      ),
+      darkTheme: themeFor(
+        isDarkTheme: true,
+        ideTheme: _ideTheme(const VsCodeTheme.dark()),
+        theme: ThemeData(useMaterial3: true, colorScheme: darkColorScheme),
+      ),
+      home: Scaffold(
+        body:
+            clientLog != null && clientDtd != null && editor != null
+                ? MockEditorWidget(
+                  editor: editor!,
+                  clientLog: clientLog!,
+                  child: PropertyEditorSidebarPanel(clientDtd!),
+                )
+                : _DtdUriForm(
+                  onSaved: _connectToDtd,
+                  formKey: GlobalKey<FormState>(),
+                ),
+      ),
+    );
+  }
+
+  /// Creates an [IdeTheme] using the colours from the mock editor.
+  IdeTheme _ideTheme(VsCodeTheme vsCodeTheme) {
+    return IdeTheme(
+      backgroundColor: vsCodeTheme.editorBackgroundColor,
+      foregroundColor: vsCodeTheme.foregroundColor,
+      embedMode: EmbedMode.embedOne,
+    );
+  }
+
+  Future<void> _connectToDtd(String? dtdUri) async {
+    if (dtdUri == null) return;
+    final uri = Uri.parse(dtdUri);
+    final connection = await createLoggedWebSocketChannel(uri);
+    setState(() {
+      clientLog = connection.log;
+      clientDtd = DartToolingDaemon.fromStreamChannel(connection.channel);
+      editor = SimulatedEditor(uri);
+    });
+  }
+}
+
+class _DtdUriForm extends StatelessWidget {
+  const _DtdUriForm({required this.onSaved, required this.formKey});
+
+  final void Function(String?) onSaved;
+  final GlobalKey<FormState> formKey;
+
+  // We assume a DTD is available on 8500. There's a VS Code task that
+  // launches this as part of the standalone_ui/editor_sidebar config.
```

Comment:
nit: I guess this should say `standalone_ui/property_editor_sidebar` config?

Is the intention that the mock editor will have the ability to send fake editor events, or that you'd just always use this with a real IDE and connect to DTD that it spawned (if the latter, the DTD being spawned on 8500 is probably pointless?).